Fluid/Ast/ForStatement.cs

```diff
@@ -147,5 +163,66 @@ public override async Task<Completion> WriteToAsync(TextWriter writer, TextEncod
 
             return Completion.Normal;
         }
+
+        private class ForLoopValue : FluidValue
+        {
+            public int Length { get; set; }
+            public int Index { get; set; }
+            public int Index0 { get; set; }
+            public int RIndex { get; set; }
+            public int RIndex0 { get; set; }
+            public bool First { get; set; }
+            public bool Last { get; set; }
+
+            public int Count => Length;
+
+            public override FluidValues Type => FluidValues.Dictionary;
+
+            public override bool Equals(FluidValue other)
+            {
+                return false;
+            }
+
+            public override bool ToBooleanValue()
+            {
+                return false;
+            }
+
+            public override double ToNumberValue()
+            {
+                return Length;
+            }
+
+            public override object ToObjectValue()
+            {
+                return null;
+            }
+
+            public override string ToStringValue()
+            {
+                return "forloop";
+            }
+
+            public override ValueTask<FluidValue> GetValueAsync(string name, TemplateContext context)
+            {
+                switch (name)
+                {
+                    case "length": return new ValueTask<FluidValue>(new NumberValue(Length));
```

Comment:
Would probably make sense to have `NumberValue.Create()` that would use lookups for common values between -1 and something, like in Jint. `NumberValue` seems to be immutable.

Reply:
Correct, and remove the default constructor to always go through a lookup?

Reply:
Create could be internal static and ensure lib only uses it, but ctr allowed for outside use and extension. Probably two versions, uint (fast array lookup) and double (slow needs to check if integral, like in Jint)